Learning_test_automation/Spotkanie6_Zadania/CurrencyConversion_vBeta/Option/Transaction.cs

```diff
@@ -40,83 +41,91 @@ private MakingDecision OwnCurrency()
-            
-            decimal amount = option.GetAmount();
-            string currencySymbol = "None!";
-            decimal sentValue = 0;
-
-            if (currencyDefault == CurrencyDefault.Dollar)
-            {
-                dollar.SentValue = amount;
-                sentValue = dollar.SentValue;
-                currencySymbol = dollar.SymbolCurrency;
-            }
-            else if (currencyDefault == CurrencyDefault.Euro)
+        private void BuyOrSellCurrency()
+        {
+            string caller = new StackTrace().GetFrame(1).GetMethod().Name;
+            bool ifBuy;
+            string transaction;
+            if (caller == "Buying")
             {
-                euro.SentValue = amount;
-                sentValue = euro.SentValue;
-                currencySymbol = euro.SymbolCurrency;
+                ifBuy = true;
+                transaction = "buy";
             }
-            else if (currencyDefault == CurrencyDefault.Czech_crown)
+            else
             {
-                czechCrown.SentValue = amount;
-                sentValue = czechCrown.SentValue;
-                currencySymbol = czechCrown.SymbolCurrency;
+                ifBuy = false;
+                transaction = "sell";
             }
 
-            Console.WriteLine($"For {amount} {currencySymbol} ({result})  you will get: {sentValue} PLN.");
-        }
+            Console.Write($"\nDear user, tell me what currency you want to {transaction}: \n");
 
-        private void BuyCurrency()
-        {         
-            Console.Write("\nDear user, tell me what currency you want to buy: \n");
-            
             CurrencyDefault currencyDefault = currencyPresentation.GetCurrencyName();
             string result = currencyDefault.ToString() == "Czech_crown" ? "Czech crown" : currencyDefault.ToString();
 
-            Console.Write($"I understand my dear user, you have chosen {result}, and how many {result}s do you need buy: ");
+            Console.Write($"I understand my dear user, you have chosen {result}, and how many {result}s do you need {transaction}: ");
 
             decimal amount = option.GetAmount();
-            string currencySymbol = "None!";
-            decimal boughtValue = 0;
 
-            if (currencyDefault == CurrencyDefault.Dollar)
-            {
-                dollar.BoughtValue = amount;
-                boughtValue = dollar.BoughtValue;
-                currencySymbol = dollar.SymbolCurrency;
-            }
-            else if (currencyDefault == CurrencyDefault.Euro)
+            currencies.Add(dollar);
+            currencies.Add(euro);
+            currencies.Add(czechCrown);
+
+            Currency selectedCurrency = currencies.FirstOrDefault(c => c.NameCurrency == currencyDefault.ToString());
+
+            if (ifBuy) Purchase(currencyDefault, amount, selectedCurrency, result);
```

Comment:
Spojrz tez na komentarz dla `NameCurrency` w klasie `Currency`.

W obu metodach `Purchase` i `Sell`  przekazujesz sporo argumentow, a praktycznie wszystko zna albo powinien znac obiekt typu `Currency`.
`currencyDefault` - to tak naprawde nazwa waluty, czesciowo masz juz to w klasie `Currency`
`amount` - to akurat od uzytkownika dostajesz, ale potrzebne jest dopiero wewnatrz metody, wiec tam lepiej dopiero to wyciagac
`selectedCurrency` - obiekt typu `Currency` to powinno zostac jak jest
`result` - w sumie to tylko manipulacja stringa, glownie dla czeskiej waluty, powinno byc obsluzone w klasie `CzechCrown`, nie wymaga wedlug mnie dodatkowego przekazywania jako argument

Reply:
Chyba mam to.